### PR TITLE
Add a Token Formatter

### PR DESCRIPTION
**Pull request by kyle-rader-msft**
This creates a `TokenFormatter` to handle Authorization header formatting and base64 encoding.

### CODE REVIEW

File: src/AzureAuth.Test/Ado/TokenFormatterTest.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    using Microsoft.Authentication.AzureAuth.Ado;
+
+    using NUnit.Framework;
+
+    internal class TokenFormatterTest
+    {
+        [Test]
+        public void Header_Bearer()
+        {
+            TokenFormatter.HeaderBearer("foobar").Should().Be("Authorization: Bearer foobar");
```
Comment (goagain) [inline src/AzureAuth.Test/Ado/TokenFormatterTest.cs:17]: Somehow, I wonder if a string extension method `.WithHeaderBearer` is more fluent.

Comment (kyle-rader-msft) [reply src/AzureAuth.Test/Ado/TokenFormatterTest.cs:17]: This is an option that might fit nicely with a recommendation Tucker had while pairing.

File: src/AzureAuth/Ado/TokenFormatting.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace Microsoft.Authentication.AzureAuth.Ado
+{
+    using System;
+    using System.Text;
+
+    /// <summary>
+    /// <see cref="string"/> extension methods for formatting tokens.
+    /// </summary>
+    public static class TokenFormatting
```
Comment (goagain) [inline src/AzureAuth/Ado/TokenFormatting.cs:12]:
```suggestion
    public static class TokenFormatting
    public static class TokenFormattingExtensions
```

For extionsion class, I would like to add `extensions` as a suffix.

**Review by goagain — APPROVED** [review]

**Review by shalinikhare27 — APPROVED** [review]

File: src/AzureAuth.Test/Ado/TokenFormatterTest.cs (hunk 1 above)
Comment (reillysiemens) [reply src/AzureAuth.Test/Ado/TokenFormatterTest.cs]: Now if only I could remember what that recommendation was... 😆

**Review by reillysiemens — APPROVED** [review]

**Review by goagain — APPROVED** [review]

**Pull request merged**